src/main/resources/hudson/plugins/favorite/project/FavoriteProjectAction/action.jelly

```diff
@@ -1,10 +1,12 @@
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:l="/lib/layout">
     <j:set var="icon" value="${action.iconClassName != null ? action.iconClassName + ' icon-md' : action.iconFileName}"/>
-    <l:task icon="${icon}" title="${action.displayName}"
-            href="${rootURL}/${action.urlName}"
-            permission="${app.READ}"
-            contextMenu="false"
-            post="true"
-    />
+    <j:if test="${icon!=null}">
+        <l:task icon="${icon}" title="${action.displayName}"
+                href="${rootURL}/${action.urlName}"
+                permission="${app.READ}"
+                contextMenu="false"
+                post="false"
```

Comment:
The reason for not using post is due to the implementation of `l:task`. It does not allow icon class attributes or the ability to switch icons. Didn't see any downside to performing a get and redirect as also figured using the `l:task` tag would be better long term in case the task/action ui classes and structure changes in the future. For all those reasons this seemed like the best compromise of function vs unnecessary complexity. 

Re-worked with your suggestions.